### ISSUE TITLE
Question about comrak and snoomark

### ISSUE THREAD

**Issue by iMrDJAi** [kivikakk/comrak#205]
From https://www.reddit.com/wiki/markdown#wiki_more_about_markdown_at_reddit

> snoomark is a downstream project of comrak, which is in turn a re-implementation of cmark-gfm, GitHub's CommonMark implementation with GitHub extensions (GFM), which is in turn downstream of cmark, the CommonMark reference implementation. snoomark itself is not (yet) open source, but changes to comrak are regularly merged into snoomark.

Is it true that changes to comrak are regularly merged into snoomark? Does contributing to comrak affect snoomark directly? I indeed want to participate here if this will improve how Markdown is being rendered on Reddit.

**Comment by kivikakk** [comment]
I'm afraid I don't know the answer to your questions! 😅 If any current snoomark maintainer wants to chime in here, go for it, but that's not a side of things I have visibility on.

**Comment by iMrDJAi** [comment]
@kivikakk I understand! I'm leaving this issue open in case a snoomark maintainer wants to answer.

One of the things that Reddit doesn't support is bidirectional text, the platform is unusable for RTL languages speakers, and this can be fixed easily by adding the attribute `dir="auto"` to block elements. I'll open a pull request soon and see if this would fix that issue on Reddit.

**Comment by kivikakk** [comment]
:v: Sweet. Once the PR is merged, I'll see if I can attract any attention at Reddit.

In order to preserve CommonMark compatibility, we'll need to add this behind an option, otherwise all the specs will fail.

**Comment by iMrDJAi** [comment]
> ✌️ Sweet. Once the PR is merged, I'll see if I can attract any attention at Reddit.

@kivikakk Thanks!

> In order to preserve CommonMark compatibility, we'll need to add this behind an option, otherwise all the specs will fail.

Sure! It's okay to add it behind an option for now. I'm willing to propose this change to the CommonMark spec anyway, and once they adopt it, it will be safe to add it without an option.

**Comment by kivikakk** [comment]
Wonderful. I'm super pleased to work towards better support for RTL languages.

**Comment by iMrDJAi** [comment]
In order to achieve that, a proper spec should be made and my draft is far from being complete as I wasn't really active in the last months. Also, I think the implementation should target cmark and cmark-gfm first before thinking about porting the changes to comrak, or trying to get attention from companies like Reddit (I don't think they are interested).

I'm closing this now.
